spec/formatting.md

```diff
@@ -29,15 +129,114 @@ the local variable takes precedence.
+   If this succeeds, assign it to the mapping with the _name_ of the _option_.
+   If this fails, do not set any value in the resolved mapping for this _option_.
+4. Call the function implementation with the following arguments:
+
+   - The current _locale_.
+   - The resolved value of the _operand_, or if there is no _operand_,
+     an empty value.
+   - The resolved mapping of _options_.
+
+   The shapes of the resolved _operand_ and _option_ values is implementation-defined.
+
+   An implementation MAY include additional arguments for the function call,
+   as long as reasonable precautions are taken to keep the function interface
+   simple and minimal, and avoid introducing potential security vulnerabilities.
+
+   As implementations MAY allow custom functions to be defined by users,
+   their access to the _formatting context_ SHOULD be minimal and read-only,
+   and their execution time SHOULD be limited.
+
+5. If the call succeeds,
+   resolve the value of the _expression_ as the result of that function call.
+   If the call fails or does not return a valid value,
+   emit a Resolution error and use a _fallback value_ for the _expression_.
+
+### Fallback Resolution
+
+The _resolution_ of an _expression_ may fail in the following cases:
+
+- A failure in the resolution of a _variable_ _operand_.
+- A failure in the resolution of a _function_ _annotation_.
+- The presence of a _reserved_ _annotation_.
+
+In each such case, and error MUST be emitted
+and a **_fallback value_** used for the _expression_.
+This value depends on the shape of the _expression_:
+
+- _expression_ with _literal_ _operand_: U+007C VERTICAL LINE `|`
+  followed by the value of the Literal,
+  and then by U+007C VERTICAL LINE `|`.
+  The same representation is used for both _quoted_ and _unquoted_ values.
+
+  > Examples: `|your horse|`, `|42|`
+
+- _expression_ with _variable_ _operand_: U+0024 DOLLAR SIGN `$`
+  followed by the _variable_ _name_ of the _operand_
+
+  > Example: `$user`
+
+- _expression_ with no _operand_:
+  the _function_ starting sigil followed by its _name_
+
+  > Examples: `:platform`, `+tag`, `-tag`
+
+- Otherwise: The U+FFFD REPLACEMENT CHARACTER `�` character
+
+_Option_ names and values are not included in the _fallback value_.
+
+When an error occurs in an _expression_ with a _variable_ _operand_
+and the _variable_ refers to a local _declaration_,
+the _fallback value_ is formatted based on the _expression_ of the _declaration_,
+rather than the _expression_ in the _selector_ or _pattern_.
+
+> For example,
+> attempting to format either of the following messages within a context that
+> does not provide for the function `:func` to be successfully resolved:
+>
+> ```
+> let $var = {|horse| :func}
+> {The value is {$var}.}
+> ```
+>
+> ```
+> let $var = {|horse|}
+> {The value is {$var :func}.}
+> ```
+>
+> would in both cases result in the _pattern_ _expression_
+> resolving to a _fallback value_ of `|horse|`.
```

Comment:
```suggestion
> yields a _fallback value_ of `|horse|`.
```

Reply:
The current phrasing is intended to refer only to the result of the resolution, rather than commenting on the result of the formatting. With "yields", this distinction is lost.